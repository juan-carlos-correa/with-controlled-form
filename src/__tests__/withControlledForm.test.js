import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormWithControlled from '../demo/FormWithControlled';

test('WithControlledForm', () => {
  const component = render(<FormWithControlled />);
  expect(toJson(component)).toMatchSnapshot();
});
