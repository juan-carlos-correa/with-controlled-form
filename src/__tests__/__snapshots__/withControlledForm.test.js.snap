// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithControlledForm render 1`] = `
<WithFormMethodsHOC>
  <Form
    cleanForm={[Function]}
    errors={
      Object {
        "email": "",
        "phoneNumber": "",
      }
    }
    handleBlur={[Function]}
    handleChange={[Function]}
    handleFocus={[Function]}
    handleSubmit={[Function]}
    isFormClean={[Function]}
    values={
      Object {
        "email": "",
        "phoneNumber": "",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          className="input"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="phoneNumber"
        >
          Phone number:
        </label>
        <input
          className="input"
          id="phoneNumber"
          name="phoneNumber"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="tel"
          value=""
        />
      </div>
      <div
        className="text-center"
      >
        <button
          className="button"
          id="send"
          type="submit"
        >
          Send
        </button>
      </div>
    </form>
  </Form>
</WithFormMethodsHOC>
`;
